### PR TITLE
fix: add termination_message_policy to container schema (closes #442)

### LINKED ISSUE #442: Feature Request: termination_message_policy
### Terraform Version
Terraform v0.11.13

### Affected Resource(s)
kubernetes_deployment

I see the TerminationMessagePolicy data type is there, but there's no implementation for use as a property (such as in deployments)

### PR DESCRIPTION
**Pull request by AeroNotix**
### Description

Adds `termination_message_policy` to the container schema.

### References

#442

### CODE REVIEW

**Comment by AeroNotix** [comment]
@radeksimko please could I have a review on this PR.

**Comment by AeroNotix** [comment]
@alexsomesan @pdecat @appilon any chance of a review?

File: kubernetes/schema_container.go [hunk 1]
```diff
@@ -517,6 +517,12 @@ func containerFields(isUpdatable, isInitContainer bool) map[string]*schema.Schem
 			Default:     "/dev/termination-log",
 			Description: "Optional: Path at which the file to which the container's termination message will be written is mounted into the container's filesystem. Message written is intended to be brief final status, such as an assertion failure message. Defaults to /dev/termination-log. Cannot be updated.",
 		},
+		"termination_message_policy": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Default:     "File",
+			Description: "Optional: Indicate how the termination message should be populated. File will use the contents of terminationMessagePath to populate the container status message on both success and failure. FallbackToLogsOnError will use the last chunk of container log output if the termination message file is empty and the container exited with an error. The log output is limited to 2048 bytes or 80 lines, whichever is smaller. Defaults to File. Cannot be updated.",
```
Comment (pdecat) [inline kubernetes/schema_container.go:524]: Validation is usually done on the accepted values:
```suggestion
			Type:         schema.TypeString,
			Optional:     true,
			ValidateFunc: validation.StringInSlice([]string{"File", "FallbackToLogsOnError"}, false),
			Default:      "File",
			Description:  "Optional: Indicate how the termination message should be populated. File will use the contents of terminationMessagePath to populate the container status message on both success and failure. FallbackToLogsOnError will use the last chunk of container log output if the termination message file is empty and the container exited with an error. The log output is limited to 2048 bytes or 80 lines, whichever is smaller. Defaults to File. Cannot be updated.",
```

**Comment by AeroNotix** [comment]
@pdecat where do the tests gop?

**Comment by pdecat** [comment]
You could probably introduce new test functions `TestAccKubernetesPod_termination_message_policy_*` to exercise the possible combinations (not set, set to `File` and set to `FallbackToLogsOnError`, like https://github.com/terraform-providers/terraform-provider-kubernetes/blob/v1.11.3/kubernetes/resource_kubernetes_pod_test.go#L294-L319

**Comment by AeroNotix** [comment]
@pdecat added tests, didn't run them - hoping your CI runs them.

**Comment by AeroNotix** [comment]
@pdecat fixed up tests, can you take a look please.

**Comment by AeroNotix** [comment]
@pdecat ping? Apologies! Would like to get this merged if possible.

File: kubernetes/schema_container.go [hunk 2]
```diff
@@ -517,6 +517,13 @@ func containerFields(isUpdatable, isInitContainer bool) map[string]*schema.Schem
 			Default:     "/dev/termination-log",
 			Description: "Optional: Path at which the file to which the container's termination message will be written is mounted into the container's filesystem. Message written is intended to be brief final status, such as an assertion failure message. Defaults to /dev/termination-log. Cannot be updated.",
 		},
+		"termination_message_policy": {
+			Type:         schema.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringInSlice([]string{"File", "FallbackToLogsOnError"}, false),
+			Default:      "File",
```
Comment (pdecat) [inline kubernetes/schema_container.go:524]: Adding new defaults will probably trigger a diff for existing configuration and people usually complain about that in non major releases. Maybe this should rather be computed.

Comment (AeroNotix) [reply kubernetes/schema_container.go:524]: Just let me know what you want.... computed seems fine as the `File` default will be specified by k8s I believe, allowing us to override to `FallbackToLogsOnError`.

**Comment by AeroNotix** [comment]
@mitchellh getting kinda sick of sending PRs to hashicorp repos with zero feedback from anyone who actually maintains the code.

File: kubernetes/schema_container.go (hunk 2 above)
Comment (AeroNotix) [reply kubernetes/schema_container.go:524]: @pdecat anyone get back to you? Any specific person we could contact here?

**Comment by AeroNotix** [comment]
@aareet please can this be looked at

**Comment by aareet** [comment]
@AeroNotix thank you for this PR. It's already in our queue for review - we appreciate your patience. [FAQ](https://github.com/hashicorp/terraform-provider-kubernetes/blob/master/_about/FAQ.md#why-isnt-my-pr-merged-yet)

**Comment by AeroNotix** [comment]
Maybe you need to give active members of the community such as @pdecat commit access, lighten your heavy workload.

File: kubernetes/schema_container.go (hunk 2 above)
Comment (jrhouston) [reply kubernetes/schema_container.go:524]: 👋 I agree with @pdecat that we should make this a `Computed` field in this case. Otherwise this PR looks good!

**Comment by AeroNotix** [comment]
@jrhouston @pdecat moved over to setting the field as computed.

File: kubernetes/resource_kubernetes_pod_test.go [hunk 3]
```diff
@@ -722,6 +722,78 @@ func TestAccKubernetesPod_config_container_startup_probe(t *testing.T) {
...
+			testAccPreCheck(t)
+		},
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckKubernetesPodDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccKubernetesTerminationMessagePolicyWithFallBackToLogsOnErr(podName, imageName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckKubernetesPodExists("kubernetes_pod.test", &confPod),
+					resource.TestCheckResourceAttr("kubernetes_pod.test", "spec.0.container.0.termination_message_policy", "FallBackToLogsOnError"),
```
Comment (aareet) [inline kubernetes/resource_kubernetes_pod_test.go:790]: This test fails for me on GKE with version 1.16 with this error. Re-testing with 1.17
```
=== RUN   TestAccKubernetesPod_termination_message_policy_default
--- PASS: TestAccKubernetesPod_termination_message_policy_default (18.44s)
=== RUN   TestAccKubernetesPod_termination_message_policy_override_as_file
--- PASS: TestAccKubernetesPod_termination_message_policy_override_as_file (10.44s)
=== RUN   TestAccKubernetesPod_termination_message_policy_override_as_fallback_to_logs_on_err
    TestAccKubernetesPod_termination_message_policy_override_as_fallback_to_logs_on_err: testing.go:654: Step 0 error: Check failed: 1 error occurred:
        	* Check 2/2 error: kubernetes_pod.test: Attribute 'spec.0.container.0.termination_message_policy' expected "FallBackToLogsOnError", got "FallbackToLogsOnError"


--- FAIL: TestAccKubernetesPod_termination_message_policy_override_as_fallback_to_logs_on_err (10.04s)
FAIL
```

Comment (aareet) [reply kubernetes/resource_kubernetes_pod_test.go:790]: Seems to be a minor typo - FallBack vs Fallback

Comment (aareet) [reply kubernetes/resource_kubernetes_pod_test.go:790]: Same failure, but the other tests also pass on 1.17
```
❯ TESTARGS="-run '^TestAccKubernetesPod_termination_message_policy'" make testacc
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test "./kubernetes" -v -run '^TestAccKubernetesPod_termination_message_policy' -timeout 120m
=== RUN   TestAccKubernetesPod_termination_message_policy_default
--- PASS: TestAccKubernetesPod_termination_message_policy_default (11.34s)
=== RUN   TestAccKubernetesPod_termination_message_policy_override_as_file
--- PASS: TestAccKubernetesPod_termination_message_policy_override_as_file (15.33s)
=== RUN   TestAccKubernetesPod_termination_message_policy_override_as_fallback_to_logs_on_err
    TestAccKubernetesPod_termination_message_policy_override_as_fallback_to_logs_on_err: testing.go:654: Step 0 error: Check failed: 1 error occurred:
        	* Check 2/2 error: kubernetes_pod.test: Attribute 'spec.0.container.0.termination_message_policy' expected "FallBackToLogsOnError", got "FallbackToLogsOnError"


--- FAIL: TestAccKubernetesPod_termination_message_policy_override_as_fallback_to_logs_on_err (22.92s)
FAIL
FAIL	github.com/terraform-providers/terraform-provider-kubernetes/kubernetes	51.639s
FAIL
make: *** [testacc] Error 1
```

**Pull request merged**